Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -155,17 +160,35 @@ public final class DiscoveryViewModel: DiscoveryViewModelType, DiscoveryViewMode
 
     self.sortsAreEnabled = self.setSortsEnabledProperty.signal.skipNil()
 
-    currentParams
-      .takePairWhen(self.sortPagerSelectedSortProperty.signal.skipNil().skipRepeats(==))
-      .observeValues {
-        AppEnvironment.current.ksrAnalytics.trackDiscoverySelectedSort(nextSort: $1, params: $0)
-      }
+    Signal.combineLatest(
+      currentParams,
+      self.sortPagerSelectedSortProperty.signal.skipNil().skipRepeats(==),
+      prevSortedPage
+    ).observeValues { currentParams, currentSortedPage, prevSortedPage in
+      AppEnvironment
+        .current
+        .ksrAnalytics
+        .trackDiscoverySelectedSort(
+          prevSort: prevSortedPage,
+          nextSort: currentSortedPage,
```

Comment:
The parameter name and argument name represent two different things which suggests to me that one of them should be renamed.

Reply:
Made some changes to naming.